astropy/units/tests/test_quantity.py

```diff
@@ -507,7 +507,7 @@ def test_non_number_type(self):
         with pytest.raises(TypeError) as exc:
             q1 + {'a': 1}
         assert exc.value.args[0].startswith(
-            "Unsupported operand type(s) for ufunc add:")
+            "operand type(s) all returned NotImplemented")
```

Comment:
What do we think about this test modification? I don't know of a way to keep this test the same and accomplish our goals with this PR.

Reply:
That really is no problem -- indeed, it is already nice enough it is still a `TypeError`!

Reply:
@WilliamJamieson, yup, no problem! Thanks for your suggestion.